cmip6_downscaling/workflows/bcsd_flow.py

```diff
@@ -7,149 +7,153 @@
     tags=['dask-resource:TASKSLOTS=1'],
     result=XpersistResult(intermediate_cache_store, serializer="xarray.zarr"),
-    target="y-full-time-" + target_naming_str,
+    target=make_coarse_obs_path,
 )
 
 return_gcm_train_full_time_task = task(
     return_gcm_train_full_time,
     tags=['dask-resource:TASKSLOTS=1'],
     result=XpersistResult(intermediate_cache_store, serializer="xarray.zarr"),
-    target="x-train-full-time-" + target_naming_str,
+    target=make_rechunked_gcm_path,
 )
 
 return_gcm_predict_rechunked_task = task(
     return_gcm_predict_rechunked,
     tags=['dask-resource:TASKSLOTS=1'],
     result=XpersistResult(intermediate_cache_store, serializer="xarray.zarr"),
-    target="x-predict-rechunked-" + target_naming_str,
+    target=make_gcm_predict_path,
 )
 
 fit_and_predict_task = task(
     fit_and_predict,
     log_stdout=True,
     result=XpersistResult(intermediate_cache_store, serializer="xarray.zarr"),
-    target="fit-and-predict-" + target_naming_str,
+    target=make_bias_corrected_path,
 )
 
 postprocess_bcsd_task = task(
     postprocess_bcsd,
     tags=['dask-resource:TASKSLOTS=1'],
     log_stdout=True,
     result=XpersistResult(results_cache_store, serializer="xarray.zarr"),
-    target="postprocess-results-" + target_naming_str,
+    target=make_bcsd_output_path,
 )
 
 
 with Flow(
-    name='bcsd',
+    name="bcsd",
     storage=runtime.storage,
     run_config=runtime.run_config,
     executor=runtime.executor,
 ) as bcsd_flow:
+    obs = Parameter("OBS")
     gcm = Parameter("GCM")
     scenario = Parameter("SCENARIO")
     train_period_start = Parameter("TRAIN_PERIOD_START")
     train_period_end = Parameter("TRAIN_PERIOD_END")
     predict_period_start = Parameter("PREDICT_PERIOD_START")
     predict_period_end = Parameter("PREDICT_PERIOD_END")
-    variable = Parameter("VARIABLE")
     latmin = Parameter("LATMIN")
     latmax = Parameter("LATMAX")
     lonmin = Parameter("LONMIN")
     lonmax = Parameter("LONMAX")
+    variable = Parameter("VARIABLE")
 
-    (
-        coarse_obs_path,
-        spatial_anomalies_path,
-        bias_corrected_path,
-        final_out_path,
-    ) = make_flow_paths_task(
-        GCM=gcm,
-        SCENARIO=scenario,
-        TRAIN_PERIOD_START=train_period_start,
-        TRAIN_PERIOD_END=train_period_end,
-        PREDICT_PERIOD_START=predict_period_start,
-        PREDICT_PERIOD_END=predict_period_end,
-        VARIABLE=variable,
-        LATMIN=latmin,
-        LATMAX=latmax,
-        LONMIN=lonmin,
-        LONMAX=lonmax,
+    gcm_grid_spec, obs_identifier, gcm_identifier = path_builder_task(
+        obs=obs,
+        gcm=gcm,
+        scenario=scenario,
+        train_period_start=train_period_start,
+        train_period_end=train_period_end,
+        predict_period_start=predict_period_start,
+        predict_period_end=predict_period_end,
+        latmin=latmin,
+        latmax=latmax,
+        lonmin=lonmin,
+        lonmax=lonmax,
```

Comment:
Thoughts on further consolidating these parameters?

`train_period_{start, end}` -> `train_period_slice = slice(train_period_start, train_period_end)`
`predict_period_{start, end}` -> `predict_period_slice = slice(predict_period_start, predict_period_end)`
`bbox` -> `predict_period_slice = (lonmin, latmin, lonmax, latmax)`  # probably should be a named tuple